src/algebra/algebra/basic.lean

```diff
@@ -1537,3 +1537,19 @@ begin
 end
 
 end submodule
+
+namespace alg_hom
+
+variables {R : Type u} {A : Type v} {B : Type w} {I : Type*}
+
+variables [comm_semiring R] [semiring A] [semiring B]
+variables [algebra R A] [algebra R B]
+
+/-- `R`-algebra homomorphism between the function spaces `I → A` and `I → B`, induced by an
+`R`-algebra homomorphism `f` between `A` and `B`. -/
+@[simps]
+def comp_left (f : A →ₐ[R] B) (I : Type*) : (I → A) →ₐ[R] (I → B) :=
+{ commutes' := λ c, by { ext, exact f.commutes' c },
```

Comment:
```suggestion
{ to_fun := λ h, f ∘ h,
  commutes' := λ c, by { ext, exact f.commutes' c },
```
For `simps`

Reply:
Sure.  I don't really know how `simps` works; do I need to make similar edits on any of the other constructions?

Reply:
Try using `@[simps?]` and it will show you the lemma it generates. The goal is to generate a lemma without a reference to `to_fun` or `comp_left` on the RHS.

Reply:
OK, I went through and inspected.  I think it's not useful to add this line for the continuous constructions, because the best expression you can give for the `to_fun` is `(⟨g, hg⟩ : C(β, γ)).comp`.  These simp-lemma will likely only be used in conjunction with other simp-lemmas, so it doesn't matter that the form on its own is ugly.

It seems like an improvement for the bare (non-topologized) constructions, so I have done it there throughout.